src/_pytest/main.py

```diff
@@ -392,7 +393,7 @@ def __init__(self, config):
         self.startdir = config.invocation_dir
         self._initialpaths = frozenset()
         # Keep track of any collected nodes in here, so we don't duplicate fixtures
-        self._node_cache = {}
+        self._node_collection_cache = {}  # type: Dict[str, List["Node"]]
```

Comment:
```suggestion
        self._node_collection_cache = {}  # type: Dict[str, List[Node]]
```

Also `-> None` then for `__init__` if it works.